consensus/qbft/src/main/java/org/hyperledger/besu/consensus/qbft/statemachine/QbftBlockHeightManagerFactory.java

```diff
@@ -75,22 +76,39 @@ public BaseQbftBlockHeightManager create(final BlockHeader parentHeader) {
     }
   }
 
+  public void isEarlyRoundChangeEnabled(final boolean isEarlyRoundChangeEnabled) {
+    this.isEarlyRoundChangeEnabled = isEarlyRoundChangeEnabled;
+  }
+
   private BaseQbftBlockHeightManager createNoOpBlockHeightManager(final BlockHeader parentHeader) {
     return new NoOpBlockHeightManager(parentHeader);
   }
 
   private BaseQbftBlockHeightManager createFullBlockHeightManager(final BlockHeader parentHeader) {
-    return new QbftBlockHeightManager(
-        parentHeader,
-        finalState,
+
+    RoundChangeManager roundChangeManager =
         new RoundChangeManager(
             BftHelpers.calculateRequiredValidatorQuorum(finalState.getValidators().size()),
             messageValidatorFactory.createRoundChangeMessageValidator(
                 parentHeader.getNumber() + 1L, parentHeader),
-            finalState.getLocalAddress()),
-        roundFactory,
-        finalState.getClock(),
-        messageValidatorFactory,
-        messageFactory);
+            finalState.getLocalAddress());
+
+    QbftBlockHeightManager qbftBlockHeightManager =
+        new QbftBlockHeightManager(
+            parentHeader,
+            finalState,
+            roundChangeManager,
+            roundFactory,
+            finalState.getClock(),
+            messageValidatorFactory,
+            messageFactory);
+
+    if (isEarlyRoundChangeEnabled) {
+      roundChangeManager.setRCQuorum(
```

Comment:
Rather than having a set method, it would be nicer to just pass in the rcQuorum into the QbftBlockHeightManager constructor above